Issue search related tests

AllanJeremy opened the pull request:
Partly fixes #997

mbicknese commented on svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:
```diff
@@ -0,0 +1,69 @@
+        .map((number) => number.textContent)
+        .filter(Boolean);
+
+      expect(issueNumbers.length).toBe(4);
```
It doesn't really hurt, but you're checking for the length to be four only to follow it up by checking whether the variable equals a 4-length array. You could drop this line and keep the same level of confidence in the code.

AllanJeremy replied:
True

AllanJeremy replied:
In the second case we are testing for the order in which stuff is displayed though @mbicknese

AllanJeremy replied:
Fixed

mbicknese commented on svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:
```diff
@@ -0,0 +1,69 @@
+        .map((comment) => comment.textContent)
+        .filter(Boolean);
+
+      expect(commentCounts.length).toBe(4);
```
Same as the previous comment.

AllanJeremy replied:
fixed

mbicknese commented on svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:
```diff
@@ -0,0 +1,69 @@
+  describe('List issues', () => {
+    it('should list all issues', () => {
+      const issues = screen.queryAllByTestId('issue-search-list-item');
+      expect(issues.length).toBe(4);
```
We could extract the 4 and make it a const at the top of the file. Maybe even base it on the length of mock issues. This way the tests won't fail whenever the number of issues change.

AllanJeremy replied:
Makes sense!

AllanJeremy replied:
Fixed

mbicknese reviewed (changes requested):
Two optimization points. Also there's linting warning (see GitHub comments) you might want to look at.

AllanJeremy commented:
Fixed all mentioned issues. Linting errors existed before PR, fixed one (unused var) but left the other two accessibility errors (can't use `#`) since I don't as yet know if they would alter any behaviour

mbicknese reviewed (approved):
⛵
